packages/block-library/src/template-part/edit/placeholder.js

```diff
@@ -28,6 +30,21 @@ export default function TemplatePartPlaceholder( {
 		templatePartId
 	);
 	const blockPatterns = useAlternativeBlockPatterns( area, clientId );
+
+	const { isBlockBasedTheme, canCreateTemplatePart } = useSelect(
+		( select ) => {
+			const { getCurrentTheme, canUser } = select( coreStore );
+			return {
+				isBlockBasedTheme: getCurrentTheme()?.is_block_theme,
+				canCreateTemplatePart: canUser( 'create', {
+					kind: 'postType',
+					name: 'wp_template_part',
+				} ),
```

Comment:
At this point, it seems like you can't modify the user caps required to create a template part, and anyone who can access this page will have `edit_theme_options` capabilities. This makes the current check redundant.

Templates endpoint permission check:

https://github.com/WordPress/wordpress-develop/blob/5a30482419f1b0bcc713a7fdee3a14afd67a1bca/src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php#L182-L198